cdap-master/src/main/java/co/cask/cdap/data/tools/DatasetSpecificationUpgrader.java

```diff
@@ -100,9 +100,11 @@ public void upgrade() throws Exception {
                 byte[] colVal = columnEntry.getValue();
                 String specEntry = Bytes.toString(colVal);
                 DatasetSpecification specification = GSON.fromJson(specEntry, DatasetSpecification.class);
-                DatasetSpecification updatedSpec = updateTTLInSpecification(specification, null);
-                colVal = Bytes.toBytes(GSON.toJson(updatedSpec));
-                put.add(familyMap.getKey(), columnMap.getKey(), timeStamp, colVal);
+                if (specification != null) {
```

Comment:
I'm trying to think about why the colVal would be either null or empty bytearray, but wasn't able to figure out a reason for that.